src/resources/Indexes/Indexes.ts

```diff
@@ -0,0 +1,58 @@
+import API from '../../APICore';
+import {PageModel} from '../BaseInterfaces';
+import Resource from '../Resource';
+import {CreateCoveoIndexModel, IndexAttributes, IndexBackupsItems, IndexStatisticsModel} from './IndexesInterface';
+
+export default class Indexes extends Resource {
+    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/indexes`;
+    static indexBackupUrl = `/rest/organizations/${API.orgPlaceholder}/indexbackups/page`;
+
+    list() {
+        return this.api.get<IndexAttributes[]>(Indexes.baseUrl);
+    }
+
+    // Same for Copy Index but making sure to pass copyFromId and machineSpec
+    create(indexModal?: CreateCoveoIndexModel) {
+        return this.api.post<{id: string}>(Indexes.baseUrl, indexModal);
+    }
+
+    delete(indexId: string) {
+        return this.api.delete(`${Indexes.baseUrl}/${indexId}`);
+    }
+
+    get(indexId: string) {
+        return this.api.get<IndexAttributes>(`${Indexes.baseUrl}/${indexId}`);
+    }
+
+    backup(indexId: string) {
+        return this.api.post<{id: string}>(`${Indexes.baseUrl}/${indexId}/backup`, {});
+    }
+
+    getBackups() {
+        return this.api.get<PageModel<IndexBackupsItems>>(`${Indexes.indexBackupUrl}`);
+    }
+
+    forceCommit(indexId: string) {
+        return this.api.post(`${Indexes.baseUrl}/${indexId}/commit`, {});
+    }
+
+    readOnly(indexId: string, isReadOnly: boolean) {
+        return this.api.put(`${Indexes.baseUrl}/${indexId}/readonly`, {isReadOnly});
```

Comment:
`isReadOnly` is a query parameter, therefore it needs to be in the url rather than the request body.
![image](https://user-images.githubusercontent.com/35579930/67499093-7ad25f00-f64e-11e9-8fa6-703e5c68817b.png)